cas/cas.h

```diff
@@ -135,6 +135,28 @@ struct m0_cas_kv_vec {
 	struct m0_cas_kv *cv_rec;
 } M0_XCA_SEQUENCE M0_XCA_DOMAIN(rpc);
 
+/**
+ * CAS record version and tombstone encoded in on-disk/on-wire format.
+ * Format:
+ *     MSB                             LSB
+ *     +----------------+----------------+
+ *     | 1 bit          | 63 bits        |
+ *     |<- tombstone -> | <- timestamp ->|
+ *     +----------------+----------------+
+ *
+ * A version comprises a physical timestamp and a tombstone flag.
+ * The timestamp is set on the client side whenever the corresponding
+ * record was supposed to be modified by PUT or DEL request.
+ * The tombstone flag is set when the corresponding record has been
+ * "logicaly" removed from the storage (although, it still exists
+ * there as a "dead" record).
+ *
+ * See ::COF_VERSIONED, ::M0_CRV_VER_NONE for details.
+ */
+struct m0_crv {
```

Comment:
Why the structure type is used here ?

Reply:
Do you mean "why don't we just use uint64_t instead?". I asked myself the same question. There are two reasons:

1. It might be error-prone. The version (the pair "tombstone+timestamp") of a key-value record is a data type. The domain of this type is ([0, 1], [1..2^63]) | (0, 0). The domain of uint64_t is just 0..UINT64_MAX. A separate data type should make the reader of this code to think about that. For example, we have uint128 and fid data types for different purposes even though they have the same size.

2. There is a chance that we need to add more information here. For example, we might want to add the fid of the originator into this data type. I doubt we ever do that, but it is much easier to extend/modify a structure than a bare uint64_t.